File: Onoicrm.Domain/Entities/Clinic.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Onoicrm.Domain.Entities;
public enum CompanyType {Clinic, Company}
public enum BookingType {Doctor, Armchair}
public class Clinic : Entity
{
    public string Caption { get; set; } = "";
    public DateTimeOffset ExpiredDate { get; set; } = DateTimeOffset.Now;
    protected override int GetClassId() =>  ClassNames.Clinic;
    public long? GroupUserProfileId { get; set; }
    [JsonIgnore] public string PasswordHash { get; set; } = "";
    public int MaxDoctorCount { get; set; }
    public int MaxChairCount { get; set; }
    public string? WappiProfileId { get; set; }
    public string? WappiToken { get; set; }
    public string? WorkStartTime { get; set; }
    public string? WorkEndTime { get; set; }
    public int BookingTimeDuration { get; set; }
    public bool UseArmchairForBooking { get; set; }
    public BookingType BookingType { get; set; } = BookingType.Doctor;
    

    public CompanyType Type { get; set; } = CompanyType.Clinic;
    [NotMapped]
    public bool HasWhatsAppAccount => !string.IsNullOrEmpty(WappiToken) && !string.IsNullOrEmpty(WappiProfileId);
}

File: Onoicrm.Front/src/modules/clinic/Clinic.ts
import {Entity} from '~/shared/entities/Entity';
export enum CompanyType {Clinic,Company}
export class Clinic extends Entity {
    public caption!: string;
    public groupUserProfile:any
    public wappiProfileId:any
    public wappiToken:any
    public workStartTime:any
    public workEndTime:any
    public bookingTimeDuration:any
    public useArmchairForBooking:any
    public bookingType:any
    public type:CompanyType = CompanyType.Clinic;
    constructor(instance?: Partial<Clinic>) {
        super();
        Object.assign(this, instance);
    }
}


